diff --git a/node_modules/react-native-google-places/ios/RNGooglePlacesViewController.m b/node_modules/react-native-google-places/ios/RNGooglePlacesViewController.m
index 902a12a..e939290 100644
--- a/node_modules/react-native-google-places/ios/RNGooglePlacesViewController.m
+++ b/node_modules/react-native-google-places/ios/RNGooglePlacesViewController.m
@@ -6,7 +6,7 @@
 #import <React/RCTUtils.h>
 #import <React/RCTLog.h>
 
-@interface RNGooglePlacesViewController ()<GMSAutocompleteViewControllerDelegate>
+@interface RNGooglePlacesViewController ()<GMSAutocompleteViewControllerDelegate, GMSPlacePickerViewControllerDelegate>
 @end
 
 @implementation RNGooglePlacesViewController
@@ -50,6 +50,13 @@
 	_reject = reject;
 
 	GMSPlacePickerConfig *config = [[GMSPlacePickerConfig alloc] initWithViewport:bounds];
+    GMSPlacePickerViewController *viewController = [[GMSPlacePickerViewController alloc] initWithConfig:config];
+    viewController.delegate = self;
+
+    UIViewController *topController = [self getTopController];
+    [topController presentViewController:viewController animated:YES completion:nil];
+
+    /*
     _placePicker = [[GMSPlacePicker alloc] initWithConfig:config];
     [_placePicker pickPlaceWithCallback:^(GMSPlace *place, NSError *error) {
         if (place) {
@@ -63,6 +70,34 @@
             _reject(@"E_USER_CANCELED", @"Search cancelled", nil);
         }
     }];
+    */
+}
+
+- (void)placePicker:(GMSPlacePickerViewController *)viewController didPickPlace:(GMSPlace *)place {
+    UIViewController *topController = [self getTopController];
+    [topController dismissViewControllerAnimated:YES completion:nil];
+
+    if (_resolve) {
+        _resolve([NSMutableDictionary dictionaryWithGMSPlace:place]);
+    }
+}
+
+- (void)placePicker:(GMSPlacePickerViewController *)viewController
+   didFailWithError:(NSError *)error {
+    UIViewController *topController = [self getTopController];
+    [topController dismissViewControllerAnimated:YES completion:nil];
+
+    // TODO: handle the error.
+    NSLog(@"Error: %@", [error description]);
+
+    _reject(@"E_PLACE_PICKER_ERROR", [error description], nil);
+}
+
+- (void)placePickerDidCancel:(GMSPlacePickerViewController *)viewController {
+    UIViewController *topController = [self getTopController];
+    [topController dismissViewControllerAnimated:YES completion:nil];
+
+    _reject(@"E_USER_CANCELED", @"Search cancelled", nil);
 }
 
 
